### ISSUE TITLE
`find_homography_dlt`: a correspondence with weight 0 still changes H through the point normalisation (0.5–1 px with 1-px noise)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4890]
`find_homography_dlt` documents per-correspondence `weights`, and a weight of 0 does remove that correspondence from the DLT equations. It does not remove it from the Hartley normalisation that precedes them: `normalize_points` centres and scales *all* points, including the zero-weight ones, so a far outlier given weight 0 changes the normalising transforms of both images and, on noisy data, the solution. On twelve exact planar matches the effect is at roundoff; with 1 px of noise, adding one far correspondence with weight 0 shifts the estimated homography by up to `0.47` px (`solver="lu"`) or `0.99` px (`solver="svd"`) over a 640 x 480 frame. Computing the normalisation from the non-zero-weight points only removes the effect (the audit measured ~1e-12 px). This is the `find_fundamental` defect of #4875 in the homography estimator; `find_homography_dlt_iterated` inherits it on every re-weighted iteration, where soft weights close to 0 are the normal case.

## Repro

```python
import torch
from kornia.geometry.homography import find_homography_dlt

D = torch.float64
# 12 planar correspondences (the batch-8 fixture: a plane at depth ~5 seen by two cameras)
p1 = torch.tensor([[[318.6412, 297.1567], [219.5427, 150.0056], [317.0893, 310.6391], [358.6205, 158.1181],
                    [168.7717, 95.2103], [326.6001, 266.9535], [215.0918, 135.9827], [483.3591, 161.5411],
                    [289.8544, 218.8074], [154.4874, 87.3121], [264.6088, 316.5891], [254.0967, 105.0011]]], dtype=D)
p2 = torch.tensor([[[386.1893, 320.6106], [268.8963, 173.1520], [381.6803, 338.8034], [428.4880, 173.8503],
                    [201.3651, 116.0530], [391.7806, 291.6549], [265.2129, 158.6236], [572.5521, 154.4192],
                    [355.6987, 240.8296], [186.9599, 108.6069], [317.9339, 342.3101], [312.1000, 121.1123]]], dtype=D)
# (any 12 exact matches under one homography give the same picture; these are proj(P1 X), proj(P2 X) for a planar X)
h = lambda p: torch.cat([p, torch.ones_like(p[..., :1])], -1)
def warp(H, p):
    q = (H @ h(p).transpose(1, 2)).transpose(1, 2)
    return q[..., :2] / q[..., 2:]
g = torch.Generator().manual_seed(0)
ys, xs = torch.meshgrid(torch.arange(0, 480, 60.0, dtype=D), torch.arange(0, 640, 80.0, dtype=D), indexing="ij")
grid = torch.stack([xs.flatten(), ys.flatten()], -1)[None]
out1 = torch.tensor([[[5000.0, -3000.0]]], dtype=D); out2 = torch.tensor([[[-4000.0, 6000.0]]], dtype=D)
for noise in (0.0, 1.0):
    p2n = p2 + torch.randn(p2.shape, generator=g, dtype=D) * noise
    for solver in ("lu", "svd"):
        H_ref = find_homography_dlt(p1, p2n, torch.ones(1, 12, dtype=D), solver=solver)
        w = torch.cat([torch.ones(1, 12, dtype=D), torch.zeros(1, 1, dtype=D)], 1)
        H_zw = find_homography_dlt(torch.cat([p1, out1], 1), torch.cat([p2n, out2], 1), w, solver=solver)
        shift = (warp(H_ref, grid) - warp(H_zw, grid)).norm(dim=-1).max().item()
        print(f"noise {noise:.0f} px, solver={solver}: max shift over a 640x480 grid from adding one zero-weight far outlier = {shift:.3e} px")
```

```
noise 0 px, solver=lu: max shift over a 640x480 grid from adding one zero-weight far outlier = 1.378e-10 px
noise 0 px, solver=svd: max shift over a 640x480 grid from adding one zero-weight far outlier = 5.329e-10 px
noise 1 px, solver=lu: max shift over a 640x480 grid from adding one zero-weight far outlier = 4.702e-01 px
noise 1 px, solver=svd: max shift over a 640x480 grid from adding one zero-weight far outlier = 9.899e-01 px
```

(The literal points above are rounded to 4 decimals; the run that printed the output used the unrounded fixture, and the rounded ones give the same orders of magnitude.) kornia `main` 7ddf731bd, torch 2.14.0, CPU float64, macOS arm64.

## Classification

Silent wrong result: a documented `weights=0` does not do what "removes the correspondence" means once the data is noisy, and the effect grows with the outlier's distance. No idiom of 1–9; candidate, shared with #4875: a weighted estimate normalised with unweighted statistics.

## Disposition

Focused fix, welcome as a PR, together with #4875: compute the Hartley normalisation from the correspondences with non-zero weight (or weight the mean and scale), in `find_homography_dlt` and `find_homography_lines_dlt`. Found by the batch-8 conventions audit (two-view geometry); the batch-8c docs PR documents it and a `test_wart_*` pin flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
